src/bridging/providers/across/AcrossApi.ts

```diff
@@ -0,0 +1,371 @@
+import { TargetChainId } from 'src/chains'
+
+const ACROSS_API_URL = 'https://app.across.to/api'
+
+export interface AvailableRoutesRequest {
+  originChainId: string
+  originToken: string
+  destinationChainId: string
+  destinationToken: string
+}
+
+export interface Route {
+  originChainId: string
+  originToken: string
+  destinationChainId: string
+  destinationToken: string
+  originTokenSymbol: string
+  destinationTokenSymbol: string
+}
+
+export interface SuggestedFeesRequest {
+  token: string
+  // inputToken: string
+  // outputToken: string
+  originChainId: TargetChainId
+  destinationChainId: TargetChainId
+
+  /**
+   * Amount of the token to transfer.
+   *
+   * Note that this amount is in the native decimals of the token. So, for WETH, this would be the amount of
+   * human-readable WETH multiplied by 1e18.
+   *
+   * For USDC, you would multiply the number of human-readable USDC by 1e6.
+   *
+   * Example: 1000000000000000000
+   */
+  amount: string
+
+  /**
+   * Recipient of the deposit. Can be an EOA or a contract. If this is an EOA and message is defined, then the API will throw a 4xx error.
+   *
+   * Example: 0xc186fA914353c44b2E33eBE05f21846F1048bEda
+   */
+  recipient?: string
+
+  /**
+   * The quote timestamp used to compute the LP fees. When bridging with across, the user only specifies the quote
+   * timestamp in their transaction. The relayer then determines the utilization at that timestamp to determine the
+   * user's fee. This timestamp must be close (within 10 minutes or so) to the current time on the chain where the
+   * user is depositing funds and it should be <= the current block timestamp on mainnet. This allows the user to know
+   * exactly what LP fee they will pay before sending the transaction.
+   *
+   * If this value isn't provided in the request, the API will assume the latest block timestamp on mainnet.
+   *
+   * Example: 1653547649
+   */
+  timestamp?: number
+
+  /**
+   * Optionally override the relayer address used to simulate the fillRelay() call that estimates the gas costs
+   * needed to fill a deposit. This simulation result impacts the returned suggested-fees. The reason to customize the
+   * EOA would be primarily if the recipientAddress is a contract and requires a certain relayer to submit the fill,
+   * or if one specific relayer has the necessary token balance to make the fill.
+   *
+   * Example: 0x428AB2BA90Eba0a4Be7aF34C9Ac451ab061AC010
+   */
+  relayer?: string
+}
+
+export interface SuggestedFeesLimits {
+  /**
+   * The minimum deposit size in the tokens' units.
+   *
+   * Note: USDC has 6 decimals, so this value would be the number of USDC multiplied by 1e6. For WETH, that would be 1e18.
+   */
+  minDeposit: string
+
+  /**
+   * The maximum deposit size in the tokens' units. Note: The formatting of this number is the same as minDeposit.
```

Comment:
I would standardize and call `token atoms`.
Units has been used to indicate the, well, units. For example, 1 unit of USDC is 1. While 1 unit of USDC in atoms is 1 * 10^6.

Reply:
okok, at this point I will resolve of the comments of API, re-open if you fee they apply to this PR